### PR TITLE
refactor(ui): Add clerk components to landing page

### PR DESCRIPTION
**Pull request by alexcarpenter**
## Description

Closes SDKI-141

Make use of `SignedIn` `SignedOut` components to render links to the different flows, along with adding the ability to easily log out with `SignOutButton`

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [x] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/ui/src/components/sign-in/sign-in.tsx [hunk 1]
```diff
@@ -226,6 +226,7 @@ export function SignInComponentLoaded() {
                             <LinkButton
                               size='sm'
                               disabled={isGlobalLoading}
+                              type='button'
```
Comment (alexcarpenter) [inline packages/ui/src/components/sign-in/sign-in.tsx:229]: Fixes issue where hitting enter button would take you to forgot password flow vs submitting the password form.

**Comment by joe-bell** [comment]
Linking SDKI-141

**Review by joe-bell — APPROVED** [review]

File: packages/ui/theme-builder/app/page.tsx [hunk 2]
```diff
@@ -1,3 +1,31 @@
...
+      </SignedIn>
+      <SignedOut>
+        <div className='flex flex-col gap-4'>
+          <SignInButton>
+            <Button>Sign in</Button>
+          </SignInButton>
+          <SignUpButton>
+            <Button>Sign up</Button>
+          </SignUpButton>
+        </div>
```
Comment (joe-bell) [inline packages/ui/theme-builder/app/page.tsx:27]: What if we replaced this with links to our views? Got a little caught off guard when it redirected to the old UI!

Comment (joe-bell) [reply packages/ui/theme-builder/app/page.tsx:27]: I'll take care of this in https://linear.app/clerk/issue/SDKI-142/theme-builder-redirect-signin-and-signup-buttons-to-newly-added

**Pull request merged**